### PR TITLE
fix: Resubmitting a file should delete previous file and columns

### LINKED ISSUE #276: [bug]: If going back from `column-review` to select new file, columns from both files are extracted and shown in `column-review`
### Steps to reproduce

1. Click "Create New Table" > Fill in dialog > Upload file > Click "Extract Datastructure" 
2. At page `column-review`, click "back"/ <-- (in your browser) 
3. Return to `file-upload` and select other file to upload > Click`Extract Datastructure` again to return to `column-review`

### Expected result

Expected result would be that the upload/extraction is "reset" when going back to `file-upload` from `column-review` so that the user can select a new file to extract column metadata from.

If the user go back from `column-review` to `file-upload`, I can't come back to `column-review` (since I can't click `Extract Datastructure` again. I also can't upload the same file again. Only if I select a new file to upload, it's uploaded and I can click `Extract Datastructure` again. 

### Actual result (the bug)

After selecting a new file in `file-upload` and clicking `Extract Datastructure`, columns from both files are included in the column review. 

I think this flow could happen, if a user realises that they have selected the wrong file when they see the columns in `column-review`. 

Alternatively, if they click back by mistake and want to continue with the file the selected first, they cannot easily return to `column-review`, since `Extract Datastructure` cannot be clicked again - unless a new file is selected for upload.

### Optional: Relevant log output

_No response_

### Severity/Priority

None

### Contact Details

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by philter87**
## Description

- This PR fixes a bug. If the user re-uploads a file on the create-metadata page, we want to delete the previous file and the associated ColumnMetadata objects. This is a likely scenario, if the user wants to adjust a csv file a after review the extracted metadata.

## Related Issues

Closes #276 

## Testing

- [X] Yes
- [ ] No, not needed (give a reason below)
- [ ] No, I need help writing them

## Reviewer Focus

- Most stuff is done in metadata_create.py. When a user creates metadata based on a file, we clear existing files and ColumnMetadata objects for that table. A user can upload a file twice if they go back from column-review to upload a new file.
- I clear the file input to ensure that the "onchange"-event is fired. If the same filename is uploaded twice, then the event is not fired

## Checklist

For all PRs that are not general documentation

- [X] Tests accompany or reflect changes to the code
- [X] Tests ran and passed locally
- [X] Ran the linter and formatter
- [X] Build has passed locally
- [X] Relevant documentation has been updated

### CODE REVIEW

**Review by signekb — COMMENTED** [review]
When I test it locally, I can't click the dropzone anymore? 🐛 

It is possible to drag a file into the dropzone and upload it though (and the file disappears when I go back from column review) 👍

**Comment by philter87** [comment]
> When I test it locally, I can't click the dropzone anymore? 🐛
> 
> It is possible to drag a file into the dropzone and upload it though (and the file disappears when I go back from column review) 👍

Weird! I am able to do it locally and in the preview environment

**Comment by philter87** [comment]
@signekb Can you show me tomorrow? How it looks on your computer?

**Comment by signekb** [comment]
> Weird! I am able to do it locally and in the preview environment

I'm not able to do it anywhere. Not even in the deployment in the readme - so it's probably not something from this PR. I'll make an issue on it 👍

File: sprout/templates/file-drop-zone.html [hunk 1]
```diff
@@ -106,4 +106,11 @@ <h5>Extracting Column Metadata</h5>
   // is reloaded, when user clicks the 'back button' in the browser
   window.addEventListener('unload', function(){});
   window.addEventListener('beforeunload', function(){});
+
+  function openFileBrowserDialog(){
+    const file_input = query('uploaded_file');
+    // input cleared to ensure onchange is fired when file name is selected twice
+    file_input.value = null;
+    file_input.click();
+  }
```
Comment (signekb) [inline sprout/templates/file-drop-zone.html:115]: I'm not sure I understand this function entirely. Do you mind elaborating a bit on what happens in each line?

File: sprout/tests/tests_views_metadata_create.py [hunk 2]
```diff
@@ -65,6 +65,24 @@ def test_upload_failed_with_no_rows_found(self):
 
         self.assertContains(response, "Invalid CSV. No rows found!")
 
+    def test_resubmit_of_file_should(self):
```
Comment (signekb) [inline sprout/tests/tests_views_metadata_create.py:68]:
```suggestion
    def test_resubmit_of_file_should_delete_prev_file_and_columns(self):
```

File: sprout/tests/tests_views_metadata_create.py
Comment (signekb) [inline sprout/tests/tests_views_metadata_create.py:1]: Do we need a test on that submitting a file, going to column-review, then going back and submitting another file, ties to the correct `tablemetadata_id` and that only the columns from the second file is included? The test `test_resubmit_file_should_...`only tests what happens when the same file is uploaded twice

**Review by signekb — COMMENTED** [review]
Nice 👌 I have a few comments/questions.

File: sprout/templates/file-drop-zone.html (hunk 1 above)
Comment (philter87) [reply sprout/templates/file-drop-zone.html]: I see.. This is a bit weird. I have added some comments both to this method and to the html

File: sprout/tests/tests_views_metadata_create.py
Comment (philter87) [reply sprout/tests/tests_views_metadata_create.py:1]: Great idea, I will adapt the test

**Review by signekb — APPROVED** [review]
👍

**Review by lwjohnst86 — APPROVED** [review]
Great catch!

**Pull request merged**